containers/performance_tests/TestDualView.hpp

```diff
@@ -0,0 +1,200 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_TEST_DUALVIEW_HPP
+#define KOKKOS_TEST_DUALVIEW_HPP
+
+#include <iostream>
+#include <cstdlib>
+#include <cstdio>
+#include <Kokkos_DualView.hpp>
+
+#include <Kokkos_Timer.hpp>
+
+namespace Performance {
+
+namespace Impl {
+
+template <typename Scalar, class ViewType>
+struct SumViewEntriesFunctor {
+  using value_type = Scalar;
+  ViewType view;
+  SumViewEntriesFunctor(const ViewType& view_) : view(view_) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i, value_type& sum) const {
+    for (size_t j = 0; j < view.extent(1); ++j) {
+      sum += view(i, j);
+    }
+  }
+};
+
+template <typename Scalar, class ViewType>
+struct IncrViewEntriesFunctor {
+  using value_type = Scalar;
+  ViewType view;
+  IncrViewEntriesFunctor(const ViewType& view_) : view(view_) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i) const {
+    for (size_t j = 0; j < view.extent(1); ++j) {
+      view(i, j)++;
+    }
+  }
+};
+
+template <typename Scalar, class Device>
+struct test_dualview_with_datacheck {
+  using scalar_type     = Scalar;
+  using execution_space = Device;
+
+  template <typename ViewType>
+  void run(const int n, const int m) {
+    ViewType a, b;
+    a = ViewType("A", n, m);
+    b = ViewType("B", n, m);
+
+    const scalar_type sum_total = scalar_type(n * m);
+
+    Kokkos::deep_copy(a.d_view, 1);
+
+    using device_space = typename ViewType::t_dev::execution_space;
+    using host_space   = typename ViewType::t_host::execution_space;
+
+    a.template modify<device_space>();
+    a.template sync<host_space>();
+
+    scalar_type a_d_sum    = 0;
+    using t_dev_exec_space = typename ViewType::t_dev::execution_space;
+
+    Kokkos::parallel_reduce(
+        Kokkos::RangePolicy<device_space>(0, n),
+        SumViewEntriesFunctor<scalar_type, typename ViewType::t_dev>(a.d_view),
+        a_d_sum);
+    ASSERT_EQ(a_d_sum, sum_total);
+
+    // Check host view is synced as expected
+    scalar_type a_h_sum = 0;
+    for (size_t i = 0; i < a.h_view.extent(0); ++i)
+      for (size_t j = 0; j < a.h_view.extent(1); ++j) {
+        a_h_sum += a.h_view(i, j);
+      }
```

Comment:
Why are you doing all these asserts and checks in a performance test? You are measuring your own loop instead of the `parallel_reduce`.

Reply:
Removed, thanks

Reply:
I have left the asserts since for a no debug run they come as `nop`.